src/loaders/OmeZarrLoader.ts

```diff
@@ -81,89 +83,125 @@ type OMEZarrMetadata = {
+      return this.baseStore.getItem(item, opts);
+    }
+
+    const { scaleNames, axesTCZYX, dimensionSeparator } = this.arrayInfo;
+
+    // Extract chunk coordinates from item path
+    const pathElems = item.split(dimensionSeparator);
+    // The first element should be the scale name - convert it to scale level index
+    const scale = scaleNames.indexOf(pathElems.shift() as string);
+    // The remaining elements should be chunk coordinates, and should be numbers
+    const coordsLength = getDimensionCount(axesTCZYX);
+    const chunkCoords = pathElems.slice(-coordsLength).map((s) => parseInt(s, 10));
+    if (chunkCoords.length < coordsLength || chunkCoords.some(isNaN) || scale < 0) {
+      console.log("CachingStore: unexpected item path: " + item);
+      return this.baseStore.getItem(item, opts);
+    }
+
+    // Check the cache
+    const [t, c, z, y, x] = axesTCZYX;
+    const time = t < 0 ? 0 : chunkCoords[t];
+    const channel = c < 0 ? 0 : chunkCoords[c];
+    const chunk = new Vector3(chunkCoords[x], chunkCoords[y], chunkCoords[z]);
+    const cacheResult = this.cache.get(this.cacheStore, channel, { scale, time, chunk });
+    if (cacheResult) {
+      return cacheResult;
+    }
 
-function imageIndexFromLoadSpec(loadSpec: LoadSpec, multiscales: OMEMultiscale[]): number {
-  // each entry of multiscales is a multiscale image.
-  let imageIndex = loadSpec.scene;
-  if (imageIndex !== 0) {
-    console.warn("WARNING: OMEZarrLoader does not support multiple scenes. Results may be invalid.");
+    // Not in cache; load the chunk and cache it
+    const result = await this.baseStore.getItem(item, opts);
+    this.cache.insert(this.cacheStore, result, { scale, time, chunk, channel });
+    return result;
   }
-  if (imageIndex >= multiscales.length) {
-    console.warn(`WARNING: OMEZarrLoader: scene ${imageIndex} is invalid. Using scene 0.`);
-    imageIndex = 0;
+
+  keys(): Promise<string[]> {
+    return this.baseStore.keys();
   }
-  return imageIndex;
-}
 
-function remapAxesToTCZYX(axes: Axis[]): [number, number, number, number, number] {
-  const axisTCZYX: [number, number, number, number, number] = [-1, -1, -1, -1, -1];
-  for (let i = 0; i < axes.length; ++i) {
-    const axis = axes[i];
-    if (axis.name === "t") {
-      axisTCZYX[0] = i;
-    } else if (axis.name === "c") {
-      axisTCZYX[1] = i;
-    } else if (axis.name === "z") {
-      axisTCZYX[2] = i;
-    } else if (axis.name === "y") {
-      axisTCZYX[3] = i;
-    } else if (axis.name === "x") {
-      axisTCZYX[4] = i;
-    } else {
-      console.log("ERROR: UNRECOGNIZED AXIS in zarr: " + axis.name);
-    }
+  setItem(item: string, value: ArrayBuffer): Promise<boolean> {
+    return this.baseStore.setItem(item, value);
   }
-  if (axisTCZYX[2] === -1 || axisTCZYX[3] === -1 || axisTCZYX[4] === -1) {
-    console.log("ERROR: zarr loader expects a z, y, and x axis.");
+
+  deleteItem(item: string): Promise<boolean> {
+    return this.baseStore.deleteItem(item);
```

Comment:
This looks scary.  Consider never deleting from baseStore and console log instead?

Reply:
I believe this and your below comment is prevented from happening by the `mode: "r"` (read-only) argument passed into `openArray`/`openGroup` and the fact that we never call any functions that begin with "set," but I also felt safe doing this because `deleteItem` is unimplemented by `HTTPStore`. Worth at least adding warnings though.